src/Numerics/Complex32.cs

```diff
@@ -184,7 +184,28 @@ public float Phase
         public float Magnitude
         {
             [TargetedPatchingOptOut("Performance critical to inline this type of method across NGen image boundaries")]
-            get { return (float)Math.Sqrt((_real * _real) + (_imag * _imag)); }
+            get
+            {
+                float a = Math.Abs(_real);
+                float b = Math.Abs(_imag);
+                if (float.IsPositiveInfinity(a) && float.IsPositiveInfinity(b))
```

Comment:
Wouldn't this even hold with an `||` instead of a `&&`, i.e. if just one of them is infinite?